Source/WebKit/webpushd/PushService.mm

```diff
@@ -55,32 +61,69 @@ static void updateTopicLists(PushServiceConnection& connection, PushDatabase& da
 
 #if HAVE(APPLE_PUSH_SERVICE_URL_TOKEN_SUPPORT)
 
+static void performAfterFirstUnlock(Function<void()>&& function)
+{
+#if PLATFORM(IOS) && USE(APPLE_INTERNAL_SDK)
+    if (MKBDeviceUnlockedSinceBoot()) {
+        function();
+        return;
+    }
+
+    RELEASE_LOG(Push, "Device is locked. Delaying init until it unlocks for the first time.");
+
+    static std::once_flag once;
+    static NeverDestroyed<Vector<Function<void()>>> functions;
+
+    std::call_once(once, [function = WTFMove(function)]() mutable {
+        int notifyToken;
+        notify_register_dispatch(kMobileKeyBagLockStatusNotifyToken, &notifyToken, dispatch_get_main_queue(), makeBlockPtr([function = WTFMove(function)](int token) mutable {
+            if (!notify_is_valid_token(token) || !MKBDeviceUnlockedSinceBoot())
+                return;
+
+            RELEASE_LOG(Push, "Device has unlocked. Running delayed initialization.");
+
+            for (auto &function : functions.get())
+                function();
+            functions->clear();
+
+            notify_cancel(token);
+        }).get());
+    });
+
+    functions->append(WTFMove(function));
```

Comment:
And this does a WTFMove() of the function.

(See above)

For the first function that comes in, that's not right, is it?
(I might be missing something)